app/controllers/application_controller.rb

```diff
@@ -1,10 +1,156 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    use Rack::Flash
+    set :session_secret, "ek_and_mk_for_lyfe"
+  end
+
+  helpers do
+    def current_user
+      User.find(session[:user_id])
+    end
+
+    def is_logged_in?(session)
+      session.has_key?(:user_id)
```

Comment:
It's better to rely on the successful call to `current_user` than only introspecting on the `session[:id]` value. In theory, a user might have an invalid `id` in their session which would make `logged_in?` return true even though that user would not be findable later. Also, because `current_user` is memoized, it makes sense to call it as early in the request cycle as possible, which generally is the first call `logged_in?`.

```ruby
def logged_in?
  !!current_user
end
```